crates/oxc_linter/src/disable_directives.rs

```diff
@@ -161,17 +159,23 @@ impl<'a> DisableDirectivesBuilder<'a> {
                             .push(DisableRuleComment { span: comment.span, rules });
                     }
                     continue;
+                } else if
+                // If the rule text starts with a dash, it's probably a typo of the correct syntax.
+                // Like `eslint-disable-lext-nine` where `text` is `-lext-nine`
+                !text.starts_with('-') &&
+                    // We should match only `eslint-disable` comments, not `eslint-disablefoo`
+                    text.starts_with(' ')
```

Comment:
This condition is logically identical to `text.starts_with(' ')`. You should be able to remove the `!text.starts_with('-')` part.